test/Table_Tests/src/Common_Table_Operations/Join/Replace_Spec.enso

```diff
@@ -0,0 +1,98 @@
+            lookup_table = table_builder [['x', [2, 1, 4, 3]], ['z', [20, 10, 40, 30]]]
+            expected = table_builder [['x', [10, 20, 30, 40, 20]], ['y', ['a', 'b', 'c', 'd', 'e']]]
+            result = table.replace lookup_table 'x' 'x' 'z'
+            result . should_equal expected
+
+        group_builder.specify "should be able to replace values via a lookup table provided as a Map" <|
+            table = table_builder [['x', [1, 2, 3, 4, 2]], ['y', ['a', 'b', 'c', 'd', 'e']]]
+            lookup_table = Map.from_vector [[2, 20], [1, 10], [4, 40], [3, 30]]
+            expected = table_builder [['x', [10, 20, 30, 40, 20]], ['y', ['a', 'b', 'c', 'd', 'e']]]
+            result = table.replace lookup_table 'x'
+            result . should_equal expected
+
+        group_builder.specify "should fail with Illegal_Argument if the specified columns do not exist" <|
+            table = table_builder [['x', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
+            lookup_table = table_builder [['x', [2, 1, 4, 3]], ['z', [20, 10, 40, 30]]]
+            table.replace lookup_table 'q' 'x' 'z' . should_fail_with Missing_Input_Columns
+            table.replace lookup_table 'x' 'q' 'z' . should_fail_with Missing_Input_Columns
+            table.replace lookup_table 'x' 'x' 'q' . should_fail_with Missing_Input_Columns
+
+        group_builder.specify "can allow unmatched rows" <|
+            table = table_builder [['x', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
+            lookup_table = table_builder [['x', [4, 3, 1]], ['z', [40, 30, 10]]]
+            expected = table_builder [['x', [10, 2, 30, 40]], ['y', ['a', 'b', 'c', 'd']]]
+            result = table.replace lookup_table 'x'
+            result . should_equal expected
+
+        group_builder.specify "fails on unmatched rows" <|
+            table = table_builder [['x', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
+            lookup_table = table_builder [['x', [4, 3, 1]], ['z', [40, 30, 10]]]
+            table.replace lookup_table 'x' allow_unmatched_rows=False . should_fail_with Unmatched_Rows_In_Lookup
+
+        group_builder.specify "fails on non-unique keys" <|
+            table = table_builder [['x', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
+            lookup_table = table_builder [['x', [2, 1, 4, 1, 3]], ['z', [20, 10, 40, 11, 30]]]
+            table.replace lookup_table 'x' . should_fail_with Non_Unique_Key
+
+        group_builder.specify "should avoid name clashes in the generated column name" <|
+            table = table_builder [['duplicate_key', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
+            lookup_table = table_builder [['x', [2, 1, 4, 3]], ['z', [20, 10, 40, 30]]]
+            expected = table_builder [['duplicate_key', [10, 20, 30, 40]], ['y', ['a', 'b', 'c', 'd']]]
+            result = table.replace lookup_table 'duplicate_key'
+            result . should_equal expected
```

Comment:
I don't understand what this test is checking. Where are these name clashes??

Reply:
Ah, so it's testing that an internal implementation edge case. OK.

I'd consider a note like
```suggestion
        group_builder.specify "should avoid name clashes in the (internally) generated column name" <|
            table = table_builder [['duplicate_key', [1, 2, 3, 4]], ['y', ['a', 'b', 'c', 'd']]]
            lookup_table = table_builder [['x', [2, 1, 4, 3]], ['z', [20, 10, 40, 30]]]
            expected = table_builder [['duplicate_key', [10, 20, 30, 40]], ['y', ['a', 'b', 'c', 'd']]]
            result = table.replace lookup_table 'duplicate_key'
            result . should_equal expected
```

Reply:
While at it - let's add a test where the lookup table has another (unused) column that has clashing name with another column in the original table. Badly implemented usage of `merge` could replace this column as well whereas `replace` should not affect columns other than the selected one (your implementation is currently correct, but to avoid regressions).

Reply:
Done and done.